src/plugins/variational_constraints/variational_constraints_engine.jl

```diff
@@ -775,6 +776,8 @@ function convert_to_bitsets(model::Model, node::NodeLabel, neighbors, constraint
     return result
 end
 
+apply_constraints!(model::Model, context::Context, constraints) = apply_constraints!(model, context, constraints, ConstraintStack())
```

Comment:
I think its only valid for `constraints::Constraints`, right?
Also we could add 
```julia
function apply_constraints!(
    model::Model,
    context::Context,
    constraint::UnspecifiedConstraints
)
    nothing 
end
```
or even remove `UnspecifiedConstraints` in the favour of `BetheFactorization`?

Reply:
I feel that `UnspecifiedConstraints` should still use the `default_constraints`, if any are specified, whereas `BetheFactorization` overrides these constraints in favour of the Bethe Factorization